src/patterns/api-multiple-lambdas.test.ts

```diff
@@ -58,6 +61,61 @@ describe("The GuApiGatewayWithLambdaByPath pattern", () => {
     expect(Template.fromStack(stack).toJSON()).toMatchSnapshot();
   });
 
+  it("should fail if lambdas share the same app", () => {
+    const stack = simpleGuStackForTesting();
+    const defaultProps = {
+      handler: "handler.ts",
+      app: "testing",
+      runtime: Runtime.NODEJS_14_X,
+    };
+    const lambdaOne = new GuLambdaFunction(stack, "lambda-one", {
+      ...defaultProps,
+      fileName: "my-app-1.zip",
+    });
+    const lambdaTwo = new GuLambdaFunction(stack, "lambda-two", {
+      ...defaultProps,
+      fileName: "my-app-2.zip",
+    });
+    const lambdaThree = new GuLambdaFunction(stack, "lambda-three", {
+      ...defaultProps,
+      fileName: "my-app-3.zip",
+    });
+    const lambdaFour = new GuLambdaFunction(stack, "lambda-four", {
+      ...defaultProps,
+      fileName: "my-app-4.zip",
+    });
+    new GuApiGatewayWithLambdaByPath(stack, {
+      app: "testing",
+      monitoringConfiguration: { noMonitoring: true },
+      targets: [
+        {
+          path: "/test",
+          httpMethod: "GET",
+          lambda: lambdaOne,
+        },
+        {
+          path: "/test",
+          httpMethod: "POST",
+          lambda: lambdaTwo,
+        },
+        {
+          path: "/test/a/long/path",
+          httpMethod: "GET",
+          lambda: lambdaThree,
+        },
+        {
+          path: "/test/api-key",
+          httpMethod: "GET",
+          lambda: lambdaFour,
+          apiKeyRequired: true,
+        },
+      ],
+    });
+    expect(() => Template.fromStack(stack)).toThrow(
+      "GuLambdaFunction must have a unique combination of app, region and stack",
```

Comment:
The message above also includes "Found duplicate: ${node.app}". Any reason to not capture that here?

Reply:
I guess not!